data/transactions/logic/assembler.go

```diff
@@ -903,14 +906,38 @@ func asmBranch(ops *OpStream, spec *OpSpec, args []string) error {
 		return ops.error("branch operation needs label argument")
 	}
 
-	ops.referToLabel(ops.pending.Len(), args[0])
+	ops.referToLabel(ops.pending.Len()+1, args[0], ops.pending.Len()+spec.Size)
 	ops.pending.WriteByte(spec.Opcode)
 	// zero bytes will get replaced with actual offset in resolveLabels()
 	ops.pending.WriteByte(0)
 	ops.pending.WriteByte(0)
 	return nil
 }
 
+func asmSwitch(ops *OpStream, spec *OpSpec, args []string) error {
+	numOffsets, err := strconv.ParseUint(args[0], 0, 64)
```

Comment:
I think the number of cases should be implicit, not written in TEAL.  So it's
`switch label0 label1`, not `switch 2 label0 label1`

The output bytecode is still all the same, it's just that the numOffsets should be implied by len(args).